src/sql/workbench/parts/notebook/models/cell.ts

```diff
@@ -217,8 +217,11 @@ export class CellModel implements ICellModel {
 			} else {
 				// TODO update source based on editor component contents
 				if (kernel.requiresConnection && !this.notebookModel.activeConnection) {
-					this.sendNotification(notificationService, Severity.Error, localize('kernelRequiresConnection', "Please select a connection to run cells for this kernel"));
-					return false;
+					let connected = await this.notebookModel.requestConnection();
+					if (!connected) {
+						this.sendNotification(notificationService, Severity.Error, localize('kernelRequiresConnection', "Please select a connection to run cells for this kernel"));
```

Comment:
Minor: Let's move the notification to the notebook model, and only show if there was no handler present. that way if someone does the following:
- click Run cell
- Gets prompted for connection
- Cancels
They don't need to see the error message. They intentionally canceled, so we can skip showing this message